libc/stdio_scanf.c

```diff
@@ -0,0 +1,3510 @@
+	char lastWord[LOREM_IPSUM_WORDLEN];
+
+	fprintf(rd, "%s", buff);
+	rewind(rd);
+	TEST_ASSERT_EQUAL_INT(4, vfscanfWrapper(rd, "%s %s %s %s", firstWord, secondWord, thirdWord, lastWord));
+	TEST_ASSERT_EQUAL_STRING("Lorem", firstWord);
+	TEST_ASSERT_EQUAL_STRING("ipsum", secondWord);
+	TEST_ASSERT_EQUAL_STRING("dolor", thirdWord);
+	TEST_ASSERT_EQUAL_STRING("sit", lastWord);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(4, fscanf(rd, "%s %s %s %s", firstWord, secondWord, thirdWord, lastWord));
+	TEST_ASSERT_EQUAL_STRING("Lorem", firstWord);
+	TEST_ASSERT_EQUAL_STRING("ipsum", secondWord);
+	TEST_ASSERT_EQUAL_STRING("dolor", thirdWord);
+	TEST_ASSERT_EQUAL_STRING("sit", lastWord);
+
+	TEST_ASSERT_EQUAL_INT(4, vsscanfWrapper(buff, "%s %s %s %s", firstWord, secondWord, thirdWord, lastWord));
+	TEST_ASSERT_EQUAL_STRING("Lorem", firstWord);
+	TEST_ASSERT_EQUAL_STRING("ipsum", secondWord);
+	TEST_ASSERT_EQUAL_STRING("dolor", thirdWord);
+	TEST_ASSERT_EQUAL_STRING("sit", lastWord);
+
+	TEST_ASSERT_EQUAL_INT(4, sscanf(buff, "%s %s %s %s", firstWord, secondWord, thirdWord, lastWord));
+	TEST_ASSERT_EQUAL_STRING("Lorem", firstWord);
+	TEST_ASSERT_EQUAL_STRING("ipsum", secondWord);
+	TEST_ASSERT_EQUAL_STRING("dolor", thirdWord);
+	TEST_ASSERT_EQUAL_STRING("sit", lastWord);
+}
+
+
+TEST(stdio_scanf_various, s_special)
+{
+	char buff[BUFF_LEN_STR] = { 0 };
+	const char *txt = "\4399\0ns";
+
+	fprintf(rd, "%s", txt);
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, vfscanfWrapper(rd, "%s", buff));
+	TEST_ASSERT_EQUAL_CHAR(txt[3], buff[3]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[4], buff[4]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[5], buff[5]);
+	TEST_ASSERT_EQUAL_STRING(txt, buff);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%s", buff));
+	TEST_ASSERT_EQUAL_CHAR(txt[3], buff[3]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[4], buff[4]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[5], buff[5]);
+	TEST_ASSERT_EQUAL_STRING(txt, buff);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanfWrapper(txt, "%s", buff));
+	TEST_ASSERT_EQUAL_CHAR(txt[3], buff[3]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[4], buff[4]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[5], buff[5]);
+	TEST_ASSERT_EQUAL_STRING(txt, buff);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(txt, "%s", buff));
+	TEST_ASSERT_EQUAL_CHAR(txt[3], buff[3]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[4], buff[4]);
+	TEST_ASSERT_NOT_EQUAL_CHAR(txt[5], buff[5]);
+	TEST_ASSERT_EQUAL_STRING(txt, buff);
+}
+
+
+TEST(stdio_scanf_various, percent)
+{
+	char buff[BUFF_LEN_STR] = "%yes";
+	char chr[BUFF_LEN_STR] = { 0 };
+
+	fprintf(rd, "%s", buff);
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, vfscanfWrapper(rd, "%% %s", chr));
+	TEST_ASSERT_EQUAL_STRING("yes", chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%% %s", chr));
+	TEST_ASSERT_EQUAL_STRING("yes", chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanfWrapper(buff, "%% %s", chr));
+	TEST_ASSERT_EQUAL_STRING("yes", chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(buff, "%% %s", chr));
+	TEST_ASSERT_EQUAL_STRING("yes", chr);
```

Comment:
Let's check, whether % does not parse not existent percents:

```suggestion
	char buff[BUFF_LEN_STR] = "%yes% --- yes";
	char correct[BUFF_LEN_STR] = { 0 };
	char wrong[BUFF_LEN_STR] = { 0 };

	fprintf(rd, "%s", buff);
	rewind(rd);

	TEST_ASSERT_EQUAL_INT(1, vfscanfWrapper(rd, "%% %s%% --- %% %s", correct, wrong));
	TEST_ASSERT_EQUAL_STRING("yes", correct);
	TEST_ASSERT_EQUAL_STRING("", wrong);
	
	// and please correct the next lines respectively

	rewind(rd);

	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%% %s", chr));
	TEST_ASSERT_EQUAL_STRING("yes", chr);

	TEST_ASSERT_EQUAL_INT(1, vsscanfWrapper(buff, "%% %s", chr));
	TEST_ASSERT_EQUAL_STRING("yes", chr);

	TEST_ASSERT_EQUAL_INT(1, sscanf(buff, "%% %s", chr));
	TEST_ASSERT_EQUAL_STRING("yes", chr);
```